src/Plugin/LoggerPlugin.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Prooph\Bundle\EventStore\Plugin;
+
+use Prooph\EventStore\ActionEventEmitterEventStore;
+use Prooph\EventStore\Plugin\Plugin;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\DataCollector;
+use Prooph\Common\Event\ActionEvent;
+
+class LoggerPlugin implements Plugin
+{
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    public function __construct(LoggerInterface $logger = null)
+    {
+        $this->logger = $logger ?? new NullLogger();
+    }
+
+    public function attachToEventStore(ActionEventEmitterEventStore $eventStore): void
+    {
+        foreach ($eventStore::ALL_EVENTS as $eventStoreEvent) {
+            $eventStore->attach($eventStoreEvent, function (ActionEvent $event) use ($eventStoreEvent, $eventStore) {
```

Comment:
store return value of eventStore->attach in property, then you can detach later on, abstract plugin class helps you to reduce boilerplate